src/pages/en/guides/migrate-to-astro/from-nextjs.mdx

```diff
@@ -29,16 +33,825 @@ When you rebuild your Next.js site in Astro, you will notice some important diff
+    ---
+    ---
+    <html>
+        <head lang="en">
+            <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+        </head>
+        <body>
+            <div class="screen">
+                <div class='screen-contents'>
+                    <slot />
+                </div>
+            </div>
+        </body>
+    </html>
+    ```
+
+3. Import the CSS (found in `_app.js`)
+
+    Next.js imports global styling via a CSS import in `_app.js`. This import is moved to Astro's layout component:
+
+    ```astro ins={0-3} title="src/layouts/Layout.astro" 
+    ---
+    import '../styles/index.css'
+    ---
+
+    <html>
+        <head lang="en">
+            <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+        </head>
+        <body>
+            <div class="screen">
+                <div class='screen-contents'>
+                    <slot />
+                </div>
+            </div>
+        </body>
+    </html>
+    ```
+### Next API data fetching to Astro
+
+`pages/index.js` fetches and displays a list of the first 151 Pokémon using [the REST PokéAPI](https://pokeapi.co/).
+
+Here's how to recreate that in `src/pages/index.astro`, replacing `getStaticProps()` with `fetch()`.
+
+**[tl/dr]:**
+1. Identify the return().
+2. Convert JSX to Astro by replacing Next or React syntax with Astro/HTML syntax.
+3. Add any needed JavaScript, props, imports.
+
+```jsx title="pages/index.js" {6-18}
+import Link from 'next/link'
+import styles from '../styles/poke-list.module.css';
+
+export default function Home({ pokemons }) {
+    return (
+        <>
+            <ul className={`plain-list ${styles.pokeList}`}>
+                {pokemons.map((pokemon) => (
+                    <li className={styles.pokemonListItem} key={pokemon.name}>
+                        <Link className={styles.pokemonContainer} as={`/pokemon/${pokemon.name}`} href="/pokemon/[name]">
+                            <p className={styles.pokemonId}>No. {pokemon.id}</p>
+                            <img className={styles.pokemonImage} src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemon.id}.png`} alt={`${pokemon.name} picture`}></img>
+                            <h2 className={styles.pokemonName}>{pokemon.name}</h2>
+                        </Link>
+                    </li>
+                ))}
+            </ul>
+        </>
+    )
+}
+
+export const getStaticProps = async () => {
+    const res = await fetch("https://pokeapi.co/api/v2/pokemon?limit=151")
+    const resJson = await res.json();
+    const pokemons = resJson.results.map(pokemon => {
+        const name = pokemon.name;
+        // https://pokeapi.co/api/v2/pokemon/1/
+        const url = pokemon.url;
+        const id = url.split("/")[url.split("/").length - 2];
+        return {
+            name,
+            url,
+            id
+        }
+    });
+    return {
+        props: {
+            pokemons,
+        },
+    }
+}
+```
+
```

Comment:
It feels like we should have something to clarify that the next subheadings will be migrating the above code from Next to Astro. I found it a bit confusing since it's one of the few places we have a heading break between Astro and Next code.

Maybe:

```suggestion

Migrate the above Next.js code to Astro by following the steps in the following subheadings.

```

Reply:
It's a fair call-out. I was struggling with *wanting* our tl/dr 1. 2. 3. steps... and then wanting to jump into step 1 right after we've just said 3, which is why I switched to headings.

I'll think a bit more about this!

Reply:
OK, so how we handled this in the Gatsby example was just showing ONE single example (the first one) so that none of the "tl/dr" ones were used, which meant we ignored this problem.  :sweat_smile: 

@crutchcorn Can you look at where the Gatsby guide ended up (only converting `layout.js` and make a similar call here: if we just took ONE file, even if that means we don't get the full app context here, and converted it. Which file is the most helpful to do? Can we cut it to one?